src/rules/no-async-wrapper-for-expected-promise.ts

```diff
@@ -0,0 +1,74 @@
+import { AST_NODE_TYPES, type TSESTree } from '@typescript-eslint/utils';
+import { createRule, parseJestFnCall } from './utils';
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      description:
+        'Disallow unnecessary async function wrapper for expected promises',
+    },
+    fixable: 'code',
+    messages: {
+      noAsyncWrapperForExpectedPromise:
+        'Rejected/resolved promises should not be wrapped in async function',
+    },
+    schema: [],
+    type: 'suggestion',
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      CallExpression(node: TSESTree.CallExpression) {
+        const jestFnCall = parseJestFnCall(node, context);
+
+        if (jestFnCall?.type !== 'expect') {
+          return;
+        }
+
+        const { parent } = jestFnCall.head.node;
+
+        if (parent?.type !== AST_NODE_TYPES.CallExpression) {
+          return;
+        }
+
+        const [awaitNode] = parent.arguments;
+
+        if (
+          (awaitNode?.type !== AST_NODE_TYPES.ArrowFunctionExpression &&
+            awaitNode?.type !== AST_NODE_TYPES.FunctionExpression) ||
```

Comment:
we've got a utility for this

```suggestion
          !awaitNode ||
          !isFunction(awaitNode) ||
```

Reply:
TIL. I'll amend in next commit